src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -593,6 +612,8 @@ public void Commit(IReleaseSpec releaseSpec, IWorldStateTracer stateTracer, bool
             {
                 FlushToTree();
             }
+
+            codeFlushTask.GetAwaiter().GetResult();
```

Comment:
This makes me a little unhappy

Reply:
Do we need to even await it at all? Can we let it run totally async?

Reply:
Need it to have completed by the next block